tests/types/otp-test.json

```diff
@@ -0,0 +1,116 @@
+{
+  "$schema": "https://packageurl.org/schemas/purl-test.schema-0.1.json",
+  "tests": [
+    {
+      "description": "Parse test for <class 'type'> PURL",
+      "test_group": "base",
+      "test_type": "parse",
+      "input": "pkg:otp/asn1@5.4.1?platform=linux&arch=amd64&repository_url=https:%2F%2Fgithub.com%2Ferlang%2Fotp&vcs_url=git%20https:%2F%2Fgithub.com%2Ferlang%2Fotp.git#src/asn1ct.erl",
+      "expected_output": {
+        "type": "otp",
+        "namespace": null,
+        "name": "asn1",
+        "version": "5.4.1",
+        "qualifiers": {
+          "platform": "linux",
+          "arch": "amd64"
+        },
+        "subpath": "src/asn1ct.erl"
+      },
+      "expected_failure": false,
+      "expected_failure_reason": null
+    },
+    {
+      "description": "Roundtrip test for <class 'type'> PURL",
+      "test_group": "base",
+      "test_type": "roundtrip",
+      "input": "pkg:otp/asn1@5.4.1?platform=linux&arch=amd64&repository_url=https:/github.com/erlang/otp&vcs_url=git%20https:/github.com/erlang/otp.git#src/asn1ct.erl",
+      "expected_output": "pkg:otp/asn1@5.4.1?platform=linux&arch=amd64&repository_url=https:%2F%2Fgithub.com%2Ferlang%2Fotp&vcs_url=git%20https:%2F%2Fgithub.com%2Ferlang%2Fotp.git#src/asn1ct.erl",
+      "expected_failure": false,
+      "expected_failure_reason": null
+    },
+    {
+      "description": "Build test  for <class 'type'> PURL",
+      "test_group": "base",
+      "test_type": "build",
+      "input": {
+        "type": "otp",
+        "namespace": null,
+        "name": "asn1",
+        "version": "5.4.1",
+        "qualifiers": {
+          "platform": "linux",
+          "arch": "amd64"
+        },
+        "subpath": "src/asn1ct.erl"
+      },
+      "expected_output": "pkg:otp/asn1@5.4.1?platform=linux&arch=amd64&repository_url=https:%2F%2Fgithub.com%2Ferlang%2Fotp&vcs_url=git%20https:%2F%2Fgithub.com%2Ferlang%2Fotp.git#src/asn1ct.erl",
+      "expected_failure": false,
```

Comment:
The PURL component input does not include the `repostory_url `or `vcs_url `information needed to build the output